### PR TITLE
Remove old long form support

### PR DESCRIPTION
**Pull request by isaacJChen**

### CODE REVIEW

File: lib/core/versions/latest/CreateOperation.ts [hunk 1]
```diff
@@ -33,53 +30,29 @@ export default class CreateOperation implements OperationModel {
...
     const operationJsonString = operationBuffer.toString();
     const operationObject = await JsonAsync.parse(operationJsonString);
-    let createOperation;
-    if (typeof operationObject.suffixData === 'string') {
-      // TODO: SIP 2 #781 deprecates this. Should be deleted when fully switched over
-      createOperation = await CreateOperation.parseObject(operationObject, operationBuffer, false);
-    } else {
-      createOperation = CreateOperation.parseJcsObject(operationObject, operationBuffer);
-    }
+    const createOperation = CreateOperation.parseJcsObject(operationObject, operationBuffer);
```
Comment (thehenrytsai) [inline lib/core/versions/latest/CreateOperation.ts:55]: Nit: consider renaming back to `parseObject` to be consistent with other operation classes.

File: tests/core/RequestHandler.spec.ts [hunk 2]
```diff
@@ -266,26 +266,7 @@ describe('RequestHandler', () => {
...
-    const httpStatus = Response.toHttpStatus(response.status);
-
-    expect(httpStatus).toEqual(200);
-    expect(response.body).toBeDefined();
-
-    validateDidReferencesInDidDocument(response.body.didDocument, longFormDid);
-  });
-
-  it('should return a resolved DID Document given a valid long-form DID with JCS format.', async () => {
+  it('should return a resolved DID Document given a valid long-form DID.', async () => {
```
Comment (thehenrytsai) [inline tests/core/RequestHandler.spec.ts:269]: Sanity: how is test on line 139 still passing?

File: lib/core/versions/latest/CreateOperation.ts [hunk 3]
```diff
@@ -121,67 +91,6 @@ export default class CreateOperation implements OperationModel {
...
-      }
-    }
-
-    const didUniqueSuffix = CreateOperation.computeDidUniqueSuffix(operationObject.suffixData);
-    return new CreateOperation(operationBuffer, didUniqueSuffix, encodedSuffixData, suffixData, encodedDelta, delta);
-  }
-
-  private static async parseSuffixData (suffixDataEncodedString: any): Promise<SuffixDataModel> {
-    if (typeof suffixDataEncodedString !== 'string') {
-      throw new SidetreeError(ErrorCode.CreateOperationSuffixDataMissingOrNotString);
```
Comment (thehenrytsai) [inline lib/core/versions/latest/CreateOperation.ts:174]: Error code no longer used anywhere, remove from `ErrorCode`, do this for all occurrences in this PR.
I thought about automating this in the error code generation script 'coz I am sure there are many, but haven't got around to it.

File: tests/core/DocumentComposer.spec.ts [hunk 4]
```diff
@@ -552,7 +552,7 @@ describe('DocumentComposer', async () => {
       expect(() => { DocumentComposer['validateDocument'](undefined); }).toThrow(expectedError);
     });
 
-    it('should throw DocumentComposerServiceNotArray if `services` is not an array', () => {
+    it('should throw DocumentComposerPatchServiceIdNotUnique if `services` is not an array', () => {
```
Comment (thehenrytsai) [inline tests/core/DocumentComposer.spec.ts:555]: Test title sound weird.

Comment (isaacJChen) [reply tests/core/DocumentComposer.spec.ts]: oops

File: tests/generators/ErrorCodeGenerator.ts [hunk 5]
```diff
@@ -4,7 +4,44 @@ import ErrorCode from '../../lib/core/versions/latest/ErrorCode';
...
-const saveLocation = path.resolve(__dirname, '../../../lib/core/versions/latest/ErrorCode.ts');
+const errorCodeFileName = 'ErrorCode.ts';
+const latestVersionPath = '../../../lib/core/versions/latest';
+const latestVersionDirectory = path.resolve(__dirname, latestVersionPath);
+const saveLocation = path.resolve(__dirname, `${latestVersionPath}/${errorCodeFileName}`);
+
+/**
+ * Returns true if ErrorCode is used in ts files
+ */
+function findErrorCodeInDicrectory (errorCode: string, path: string): boolean {
```
Comment (thehenrytsai) [inline tests/generators/ErrorCodeGenerator.ts:15]: Nit:
```suggestion
function isErrorCodeReferencedInDicrectory (errorCode: string, path: string): boolean {
```

**Review by thehenrytsai — APPROVED** [review]

**Comment by isaacJChen** [comment]
https://github.com/decentralized-identity/sidetree/issues/849

**Pull request merged**